objects/src/builder/mod.rs

```diff
@@ -0,0 +1,253 @@
+        self.items.push(item);
+        self
+    }
+
+    pub fn build(&self) -> AccountStorage {
+        AccountStorage::new(self.items.clone(), MerkleStore::new()).unwrap()
+    }
+}
+
+/// Builder for an `AccountId`, the builder can be configured and used multipled times.
+pub struct AccountIdBuilder<T: Rng> {
+    account_type: AccountType,
+    on_chain: bool,
+    code_root: Digest,
+    storage_root: Digest,
+    rng: T,
+}
+
+impl<T: Rng> AccountIdBuilder<T> {
+    pub fn new(rng: T) -> Self {
+        Self {
+            account_type: AccountType::RegularAccountUpdatableCode,
+            on_chain: false,
+            code_root: Digest::default(),
+            storage_root: Digest::default(),
+            rng,
+        }
+    }
+
+    pub fn account_type(&mut self, account_type: AccountType) -> &mut Self {
+        self.account_type = account_type;
+        self
+    }
+
+    pub fn on_chain(&mut self, on_chain: bool) -> &mut Self {
+        self.on_chain = on_chain;
+        self
+    }
+
+    pub fn code_root(&mut self, code_root: Digest) -> &mut Self {
+        self.code_root = code_root;
+        self
+    }
+
+    pub fn storage_root(&mut self, storage_root: Digest) -> &mut Self {
+        self.storage_root = storage_root;
+        self
+    }
+
+    pub fn build(&mut self) -> Result<AccountId, AccountError> {
+        let init_seed: [u8; 32] = self.rng.gen();
+        let seed = AccountId::get_account_seed(
+            init_seed,
+            self.account_type,
+            self.on_chain,
+            self.code_root,
+            self.storage_root,
+        )?;
+
+        AccountId::new(seed, self.code_root, self.storage_root)
+    }
+}
+
+/// Builder for an `FungibleAsset`, the builder can be configured and used multipled times.
+pub struct FungibleAssetBuilder {
+    asset: AccountId,
+    amount: u64,
+}
+
+impl FungibleAssetBuilder {
+    pub const DEFAULT_AMOUNT: u64 = 10;
+
+    pub fn new(asset: AccountId) -> Result<Self, AssetError> {
+        if !matches!(asset.account_type(), AccountType::FungibleFaucet) {
+            return Err(AssetError::not_a_fungible_faucet_id(asset));
+        }
+
+        Ok(Self {
+            asset,
+            amount: Self::DEFAULT_AMOUNT,
+        })
+    }
+
+    pub fn amount(&mut self, amount: u64) -> Result<&mut Self, AssetError> {
+        if amount > FungibleAsset::MAX_AMOUNT {
+            return Err(AssetError::amount_too_big(amount));
+        }
+
+        self.amount = amount;
+        Ok(self)
+    }
+
+    pub fn with_amount(&self, amount: u64) -> Result<FungibleAsset, AssetError> {
+        FungibleAsset::new(self.asset, amount)
+    }
+
+    pub fn build_fungible(&self) -> Result<FungibleAsset, AssetError> {
+        FungibleAsset::new(self.asset, self.amount)
+    }
+}
+
+/// Builder for an `NonFungibleAssetDetails`, the builder can be configured and used multipled times.
+pub struct NonFungibleAssetDetailsBuilder<T: Rng> {
+    asset: AccountId,
+    rng: T,
+}
```

Comment:
Should we change `asset` here to `faucet_id`? I think this may be more consistent with how we name it in other places.

Reply:
renamed